### PR TITLE
 skill Taxonomy API and Board, Class Update API

### PR DESCRIPTION
**Pull request by vishwab1**
1.Made all in to camel case for folder structure master api, tenant api.
2.Added the skill taxonomy data api
3. Added the update api for board, skill, class, subskill

### CODE REVIEW

**Comment by vishwab1** [comment]
@HarishGangula  please review the skill taxonomy api and master update api

File: src/controllers/boardUpdate/updateBoardValidationSchema.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+              "type": "array",
+              "items": {
+                "type": "integer"
+              }
+            }
+          },
+          "required": ["ids"],
+          "additionalProperties": false
+        },
+        "skill_taxonomy_ids": {
```
Comment (HarishGangula) [inline src/controllers/boardUpdate/updateBoardValidationSchema.json:60]: This is the single skill taxonomy id

File: src/controllers/boardUpdate/updateBoardValidationSchema.json [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            "ids": {
+              "type": "array",
+              "items": {
+                "type": "integer"
+              },
+              "minItems": 1,
+              "uniqueItems": true
+            },
+            "l1_skill_ids": {
+              "type": "array",
```
Comment (HarishGangula) [inline src/controllers/boardUpdate/updateBoardValidationSchema.json:51]: Add the minItems and uniqueItems and remove the condition in code

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: src/routes/router.ts [hunk 3]
```diff
@@ -1,31 +1,49 @@
...
+
+router.post('/class/update/:class_id', setDataToRequestObject('api.class.update'), updateClass);
+
+router.post('/skill/update/:skill_id', setDataToRequestObject('api.skill.update'), updateSkill);
+
+router.post('/subSkill/update/:sub_skill_id', setDataToRequestObject('api.subskill.update'), updateSubSkill);
+
+router.post('/skillTaxonomy/create', setDataToRequestObject('api.skillTaxonomy.create'), createSkillTaxonomy);
+
+router.post('/skillTaxonomy/search', setDataToRequestObject('api.skillTaxonomy.search'), searchSkillTaxonomies);
```
[the remark this answers is not in the document]
Comment (vishwab1) [reply src/routes/router.ts]: changed

File: src/controllers/boardUpdate/updateBoardValidationSchema.json (hunk 2 above)
Comment (vishwab1) [reply src/controllers/boardUpdate/updateBoardValidationSchema.json:47]: Added the  minItems and uniqueItems

**Review by HarishGangula — APPROVED** [review]

**Pull request merged**